### ISSUE TITLE
access_token API: error refreshing token

### ISSUE THREAD

**Issue by gonzalad** [pingidentity/lua-resty-openidc#59]
Hello, 

Sorry, this is an error I introduced when creating the access_token API.

Whenever I call access_token (with an expired accessToken), the lua module tries to refresh the token, but without having initialized the token_endpoint URL.

I get a :
```
2017/05/28 20:06:42 [debug] 7#7: *13 [lua] openidc.lua:563: access_token(): refreshing expired access_token: 3c68223ff0994bcfb633776dc47847 with: 8bc1d56b7bad2b4b91f516aa6c0caf1
2017/05/28 20:06:42 [debug] 7#7: *13 [lua] openidc.lua:227: openidc_call_token_endpoint(): client_secret_basic: authorization header 'Basic bTJEaXcwUkxxSXF0aHc6REtxeURjQXVyZ0Jpc2NDYnVSUUZmZw=='
2017/05/28 20:06:42 [debug] 7#7: *13 [lua] openidc.lua:236: openidc_call_token_endpoint(): request body for token endpoint call: refresh_token=8bc1d56b7bad2b4b91f516aa6c0caf1&grant_type=refresh_token&scope=openid%20refreshToken
2017/05/28 20:06:42 [error] 7#7: *13 lua entry thread aborted: runtime error: /usr/local/openresty/luajit/share/lua/5.1/resty/http.lua:191: bad argument #1 to 'ngx_re_match' (string expected, got nil)
stack traceback:
coroutine 0:
 [C]: in function 'ngx_re_match'
 /usr/local/openresty/luajit/share/lua/5.1/resty/http.lua:191: in function 'parse_uri'
 /usr/local/openresty/luajit/share/lua/5.1/resty/http.lua:720: in function 'request_uri'
 /opt/lua-resty-openidc/lib/resty/openidc.lua:239: in function 'openidc_call_token_endpoint'
 /opt/lua-resty-openidc/lib/resty/openidc.lua:571: in function 'access_token'
 access_by_lua(nginx.conf:146):21: in function <access_by_lua(nginx.conf:146):1>, client: 172.31.0.1, server: localhost, request: "GET /api HTTP/1.1", host: "iam"
172.31.0.1 - - [28/May/2017:20:06:42 +0000] "GET /api HTTP/1.1" 500 541 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/58.0.3029.110 Chrome/58.0.3029.110 Safari/537.36"
```

nginx.conf extract:
```
    location / {

      access_by_lua '

      local opts = {
        redirect_uri_path = "/redirect_uri",
        discovery = "http://iam:9080/oidc/.well-known/openid-configuration",
        client_id = "m2Diw0RLqIqthw",
        client_secret = "DKqyDcAurgBiscCbuRQFfg",
        scope = "openid refreshToken",
        iat_slack = 600,
        redirect_uri_scheme = "http",
        logout_path = "/api/logout",
        token_endpoint_auth_method = "client_secret_basic",
        ssl_verify = "no"
      }

      local res, err = require("resty.openidc").authenticate(opts)

      if err then
      ngx.status = 500
      ngx.say(err)
      ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
      end
      ';

      root   html;
      index  index.html index.htm;
    }


    location /api {

      access_by_lua '

      local opts = {
        redirect_uri_path = "/redirect_uri",
        discovery = "http://iam:9080/oidc/.well-known/openid-configuration",
        client_id = "m2Diw0RLqIqthw",
        client_secret = "DKqyDcAurgBiscCbuRQFfg",
        scope = "openid refreshToken",
        iat_slack = 600,
        redirect_uri_scheme = "http",
        logout_path = "/api/logout",
        token_endpoint_auth_method = "client_secret_basic",
        ssl_verify = "no"
      }

      local access_token, err = require("resty.openidc").access_token(opts)

      if err then
      ngx.status = 401
      ngx.say(err)
      ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
      end

      -- set Authorization header to AT
      if access_token then
      ngx.req.set_header("Authorization", "Bearer " .. access_token)
      end
      ';

      proxy_pass http://192.168.0.13:10001/;
    }
```

With the previous configuration, the problem happens whenever I call http://iam/api and the token is expired.

**Comment by gonzalad** [comment]
I've the fix ready, but I'd like if possible 2 days to test it.

**Comment by zandbelt** [comment]
Sounds good, let me know

**Comment by gonzalad** [comment]
Hi Hans, I've made a fix here #60
